classes/Attachment.php

```diff
@@ -239,25 +289,63 @@ public static function attachToProduct($idProduct, $array)
      */
     public static function getProductAttached($idLang, $list)
     {
+        if (!is_array($list)) {
+            return false;
+        }
+
         $idsAttachments = [];
-        if (is_array($list)) {
-            foreach ($list as $attachment) {
-                $idsAttachments[] = $attachment['id_attachment'];
-            }
+        foreach ($list as $attachment) {
+            $idsAttachments[] = $attachment['id_attachment'];
+        }
 
-            $sql = 'SELECT * FROM `' . _DB_PREFIX_ . 'product_attachment` pa
+        $sql = 'SELECT * FROM `' . _DB_PREFIX_ . 'product_attachment` pa
 					LEFT JOIN `' . _DB_PREFIX_ . 'product_lang` pl ON (pa.`id_product` = pl.`id_product`' . Shop::addSqlRestrictionOnLang('pl') . ')
 					WHERE `id_attachment` IN (' . implode(',', array_map('intval', $idsAttachments)) . ')
-						AND pl.`id_lang` = ' . (int) $idLang;
-            $tmp = Db::getInstance()->executeS($sql);
-            $productAttachments = [];
-            foreach ($tmp as $t) {
-                $productAttachments[$t['id_attachment']][] = $t['name'];
-            }
+					AND pl.`id_lang` = ' . (int) $idLang;
+        $tmp = Db::getInstance()->executeS($sql);
+        $productAttachments = [];
+        foreach ($tmp as $t) {
+            $productAttachments[$t['id_attachment']][] = $t['name'];
+        }
+
+        return $productAttachments;
+    }
+
+    /**
+     * Get attachment products ids of current attachment for association.
+     *
+     * @return array
+     */
+    public function getWsProducts()
+    {
+        $result = Db::getInstance()->executeS('SELECT p.`id_product` AS id
+			FROM `' . _DB_PREFIX_ . 'product_attachment` pa
+			LEFT JOIN `' . _DB_PREFIX_ . 'product` p ON (p.id_product = pa.id_product)
```

Comment:
Shouldn't this be an INNER join?
```suggestion
			INNER JOIN `' . _DB_PREFIX_ . 'product` p ON (p.id_product = pa.id_product)
```